src/main/cpp/IntakeBehaviour.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2023-2024 CurtinFRC
+// Open Source Software, you can modify it according to the terms
+// of the MIT License at the root of this project
+
+#include "IntakeBehaviour.h"
+
+#include <frc/XboxController.h>
+
+IntakeManualControl::IntakeManualControl(Intake* intake, frc::XboxController& codriver) : _intake(intake), _codriver(codriver) {
+  Controls(intake);
+}
+
+void IntakeManualControl::OnTick(units::second_t dt) {
+
+  if (_codriver.GetBButtonReleased()) {
+    if (_rawControl) {
+      _intake->SetBehaviourState(IntakeBehaviourState::kIdleing);
```

Comment:
whats the point of having a Intake Behaviour State if you are also setting the subsystem state here? you probably just want to be setting the behaviour state then the subsystem state gets set from there